desktop_version/CMakeLists.txt

```diff
@@ -101,6 +105,17 @@ ELSE()
 	ADD_EXECUTABLE(vvvvvv ${VVV_SRC})
 ENDIF()
 
+# Build options
+IF(ENABLE_WARNINGS)
+	# The weird syntax is due to CMake generator expressions.
+	# Saves quite a few lines and boilerplate at the price of readability.
+	TARGET_COMPILE_OPTIONS(vvvvvv PRIVATE
+		$<$<OR:$<CXX_COMPILER_ID:GNU>,$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>>:
+			-Wall $<$<BOOL:ENABLE_WERROR>:-Werror>>
+		$<$<CXX_COMPILER_ID:MSVC>:
+			/W4 $<$<BOOL:ENABLE_WERROR>:/WX>>)
```

Comment:
This weird magic comes from https://foonathan.net/2018/10/cmake-warnings/

It confirms my feeling that CMake has one of the most unreadable syntaxes, but at least it seems to work well :)

Only tested on Linux / GCC 8 though.